### PR TITLE
Steps to add custom field in payment

### LINKED ISSUE #8927: Add custom input field in offline payment method
# New topic request

## Description

Here I found many customizations on the checkout page.
But I have one suggestion to add one more topic to this.

New Topic: Add Custom Input Field in the offline payment method

Means in the purchase order payment method we must add purchase order number. if any customer wants to add more field with the purchase order number at that time these docs will help,

https://devdocs.magento.com/guides/v2.3/howdoi/checkout/checkout_overview.html

### PR DESCRIPTION
**Pull request by ajithkumar-maragathavel**
## Purpose of this pull request

This pull request (PR) adds a new page that explains adding a custom input field on the payment section on the checkout payment step.

## Affected DevDocs pages

-  New Topic

## Links to Magento source code

-  https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Checkout/Controller/Onepage/Success.php#L30

## Related Pull Requests

- https://github.com/magento/devdocs/pull/8931

Fixes #8927 

whatsnew
Added the topic [Add a custom field for an offline payment method](https://devdocs.magento.com/guides/v2.4/howdoi/checkout/checkout_payment_new_field.html) to the Customize Checkout tutorial.

### CODE REVIEW

**Review by BarnyShergold — CHANGES_REQUESTED** [review]
Before I review fully, there were two points that were on the old PR that were not addressed.

**POINT 1**
Not sure if Steps 1 & 2 need to be this detailed. Just say create Learning/CustomField and register it. Perhaps a link to : https://devdocs.magento.com/videos/fundamentals/create-a-new-module/

**POINT 2**
This is now a deprecated method. This should be done with a Patch / Declarative Schema : https://devdocs.magento.com/guides/v2.3/extension-dev-guide/declarative-schema/

Whilst not complusory for 2.3 it is now the accepted way and would also apply to the v2.4 of this guide and futureproof the content.

Also database changes should not be done using InstallData, it should be done by a InstallSchema - you have mixed the two.

@dobooth - can we get a second opinion on this? As a 2.4v is needed and 2.3 support it, should this not be done using the declarative schema method?

Please can these be addressed before a detailed check is done?

**Comment by ajithkumar-maragathavel** [comment]
@BarnyShergold I will fix the **Point 1** and update the PR.
Let me wait for the comment from @dobooth and provide the proper fix for the **Point 2** later.

**Comment by dobooth** [comment]
Thanks for the discussion, gents. We should use the declarative schema. We certainly don't want to add new content that we know will be out of date. Let's do it properly. We can help if needed.

**Comment by ajithkumar-maragathavel** [comment]
@BarnyShergold @dobooth Updated the doc with the declarative schema approach. Please review it.

**Comment by ajithkumar-maragathavel** [comment]
@BarnyShergold @dobooth Updated the doc with the declarative schema approach. Please review it.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:68]: Delete. Also, note that this collapsible block would have hidden step 3.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:76]: Delete.  Again, this is a small sample.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:113]: Delete

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:71]: Provide a bulleted list of the files that need to overridden. Add a head3 section for each file.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:115]: Explain why this file needs to be overridden

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:241]:
```suggestion
Create an Observer file to save the custom field data to the order. For the Observer file, an `event.xml` file is required to call the observer for a particular event. For this example, the `checkout_onepage_controller_success_action` event is used.
```

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:318]:
```suggestion
Add the product to the cart and go to the checkout page and select the `purchase order` payment method. Check that the new Purchase Order Comment field is visible.
```

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:320]: Making sure that purchase orders are enabled should be a prerequisite task. Move this info to near the beginning of this topic.

```suggestion
If the `purchase order` payment method cannot be seen on the checkout page, then go to Magento Admin and enable the payment method. See **Stores** > **Settings** > **Configuration** > **Sales** > **Payment Methods** > **Other Payment Methods** > **Purchase Order**.
```

**Review by keharper — COMMENTED** [review]
@ajithkumar-maragathavel This is a great start. It'll probably take 3 or 4 rounds of comments to finalize this PR.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: Modified. Also, modify the _menu_order_ value

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:14]: Description added about the topic.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: Got it. Updated each line item with the relevant links.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: Done.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: Modified.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: Done.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:76]: Modified. Also, added the head3 section.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: Provided an explanation.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: Moved this info to the beginning of the page.

**Comment by ajithkumar-maragathavel** [comment]
@keharper Thanks for your time for the detailed review. I have updated the requested changes.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:323]: Go to the storefront as a guest user and add the product to the cart. Go to the checkout page and select the **Purchase Order** payment. Verify that the **Purchase Order Comment** field is visible. 


Should the reader add some text and verify that it all worked. Also maybe check that the value is stored in the DB?

**Review by keharper — CHANGES_REQUESTED** [review]
Apologies for taking this long to respond to your previous round of changes.

**Comment by keharper** [comment]
Forgot to tag @ajithkumar-maragathavel

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:321]: @keharper 
Yes, since we are creating a new module, we need to run the following commands:

```
bin/magento setup:upgrade
bin/magento setup:di:compile
bin/magento setup:static-content:deploy
```

Do you want me to add these commands as well before the verify step?

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: @keharper The user can add some text and place an order using **Purchase Order**  payment. Then the user can able to check on the database side to verify the value is properly is set in the database.

Do you want me to add that information as well on this step?

**Comment by ajithkumar-maragathavel** [comment]
@keharper Thanks for the suggestions. I updated most of the changes requested and updated the PR.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:321]: Yes

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: Yes

Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md]: Thanks for the suggestion. Updated the information.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:321]: Added...

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (keharper) [inline src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:346]: Each sentence should be its own numbered step.

**Review by keharper — CHANGES_REQUESTED** [review]
@ajithkumar-maragathavel There are still quite a few comments from my previous review that have not been addressed. I'll only comment on the new material here.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:348]: @keharper Updated.

File: src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md
Comment (ajithkumar-maragathavel) [reply src/guides/v2.3/howdoi/checkout/checkout_payment_new_field.md:70]: Updated..

**Comment by dobooth** [comment]
@ajithkumar-maragathavel How's this one coming along?

**Comment by ajithkumar-maragathavel** [comment]
> @ajithkumar-maragathavel How's this one coming along?

hey @dobooth yep it's coming good so far. Had couple of rounds of review for this PR. 
I have finished all the requested changes. waiting for the further review on the changes.

**Review by keharper — APPROVED** [review]
Thanks for your patience, @ajithkumar-maragathavel. From my perspective, this PR is ready to be merged. @BarnyShergold, any further comments?

**Review by BarnyShergold — APPROVED** [review]
This is an awesome piece of work - looks super clear and easy to follow.

Gold ⭐  !

**Comment by dobooth** [comment]
Great teamwork here. Thanks for all the effort!

**Comment by dobooth** [comment]
@magento import pr to magento-devdocs/devdocs

**Comment by magento-engcom-team** [comment]
@dobooth the pull request successfully imported.

**Pull request merged**
